app/src/main/java/com/termux/app/RunCommandService.java

```diff
@@ -71,125 +72,63 @@ public int onStartCommand(Intent intent, int flags, int startId) {
 
         String executableExtra = executionCommand.executable = IntentUtils.getStringExtraIfSet(intent, RUN_COMMAND_SERVICE.EXTRA_COMMAND_PATH, null);
         executionCommand.arguments = IntentUtils.getStringArrayExtraIfSet(intent, RUN_COMMAND_SERVICE.EXTRA_ARGUMENTS, null);
 
-        /*
-        * If intent was sent with `am` command, then normal comma characters may have been replaced
-        * with alternate characters if a normal comma existed in an argument itself to prevent it
-        * splitting into multiple arguments by `am` command.
-        * If `tudo` or `sudo` are used, then simply using their `-r` and `--comma-alternative` command
-        * options can be used without passing the below extras, but native supports is helpful if
-        * they are not being used.
-        * https://github.com/agnostic-apollo/tudo#passing-arguments-using-run_command-intent
-        * https://android.googlesource.com/platform/frameworks/base/+/21bdaf1/cmds/am/src/com/android/commands/am/Am.java#572
-        */
-        boolean replaceCommaAlternativeCharsInArguments = intent.getBooleanExtra(RUN_COMMAND_SERVICE.EXTRA_REPLACE_COMMA_ALTERNATIVE_CHARS_IN_ARGUMENTS, false);
-        if (replaceCommaAlternativeCharsInArguments) {
-            String commaAlternativeCharsInArguments = IntentUtils.getStringExtraIfSet(intent, RUN_COMMAND_SERVICE.EXTRA_COMMA_ALTERNATIVE_CHARS_IN_ARGUMENTS, null);
-            if (commaAlternativeCharsInArguments == null)
-                commaAlternativeCharsInArguments = TermuxConstants.COMMA_ALTERNATIVE;
-            // Replace any commaAlternativeCharsInArguments characters with normal commas
-            DataUtils.replaceSubStringsInStringArrayItems(executionCommand.arguments, commaAlternativeCharsInArguments, TermuxConstants.COMMA_NORMAL);
-        }
+        replaceComma2Char(intent, executionCommand);
 
         executionCommand.stdin = IntentUtils.getStringExtraIfSet(intent, RUN_COMMAND_SERVICE.EXTRA_STDIN, null);
         executionCommand.workingDirectory = IntentUtils.getStringExtraIfSet(intent, RUN_COMMAND_SERVICE.EXTRA_WORKDIR, null);
 
-        // If EXTRA_RUNNER is passed, use that, otherwise check EXTRA_BACKGROUND and default to Runner.TERMINAL_SESSION
-        executionCommand.runner = IntentUtils.getStringExtraIfSet(intent, RUN_COMMAND_SERVICE.EXTRA_RUNNER,
-            (intent.getBooleanExtra(RUN_COMMAND_SERVICE.EXTRA_BACKGROUND, false) ? Runner.APP_SHELL.getName() : Runner.TERMINAL_SESSION.getName()));
-        if (Runner.runnerOf(executionCommand.runner) == null) {
-            errmsg = this.getString(R.string.error_run_command_service_invalid_execution_command_runner, executionCommand.runner);
-            executionCommand.setStateFailed(Errno.ERRNO_FAILED.getCode(), errmsg);
-            TermuxPluginUtils.processPluginExecutionCommandError(this, LOG_TAG, executionCommand, false);
-            return stopService();
-        }
+        Integer invalidRunner = checkRunner(intent, executionCommand);
```

Comment:
Could just return `false` from function and do

`if(!checkRunner()) return stopService()`